File: 프로그래머스/2/42842. 카펫/카펫.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(int brown, int yellow) {
    vector<int> answer;
    for(int i=1; i*i<=yellow; i++){
        if(yellow % i > 0) continue;
        else{
            int w = yellow / i;
            int h = i;
            if(2*w+2*h+4==brown){
                answer.push_back(w+2);
                answer.push_back(h+2);
                break;
            }
        }
    }
    return answer;
}

File: 프로그래머스/2/42842. 카펫/카펫.java
class Solution {
    public int[] solution(int brown, int yellow) {
        int[] answer = new int[2];
        
        int check = brown/2 +2;
        int total = brown + yellow;
        
        int limit = (int)Math.floor(Math.sqrt(total));
        for(int i=3; i<= limit; i++){
            if(total % i == 0){
                if((total/i) + i == check){
                    answer[0] = total/i;
                    answer[1] = i;
                    break;
                }
            }
        }
        
        return answer;
    }
}